tunnel/ssh_keys.go

```diff
@@ -0,0 +1,54 @@
+package tunnel
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"syscall"
+
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/crypto/ssh/terminal"
+)
+
+func privateKeyFile(path string) (ssh.AuthMethod, error) {
+	log.Debugf("Parsing privatekey %s", path)
+
+	buffer, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, errors.New("cannot read SSH key file " + path)
+	}
+	if len(buffer) == 0 {
+		return nil, errors.New("bad key file empty file")
+	}
+
+	block, _ := pem.Decode(buffer)
+	if block == nil {
+		return nil, errors.New("bad key file")
+	}
+
+	// Return early if the SSH file is not password protected
+	if !x509.IsEncryptedPEMBlock(block) {
+		key, errParse := ssh.ParsePrivateKey(buffer)
+		if errParse != nil {
+			return nil, errors.New("cannot parse SSH key file " + path)
+		}
+		return ssh.PublicKeys(key), nil
+	}
+
+	fmt.Println("Your password: ")
+
+	bytePassword, err := terminal.ReadPassword(int(syscall.Stdin))
+	if err != nil {
+		return nil, errors.New("could not read your password " + err.Error())
+	}
+
+	key, err := ssh.ParsePrivateKeyWithPassphrase(buffer, bytePassword)
+	if err != nil {
+		return nil, errors.New("cannot parse SSH key file " + path)
+	}
+
+	return ssh.PublicKeys(key), nil
```

Comment:
Are we returning plural keys here? If so I think the variable name in `tunnel.go` should reflect that.